static/js/whyis_vue/modules/events/functions.js

```diff
@@ -0,0 +1,278 @@
+/**
+ * The functions below are used by the event services module
+ * 1. "confirmAuth": Checks if user is authenticated
+ * 2. "toggleNav"
+ * current_user.has_role('Admin')
+ */
+
+
+import { getViewUrl } from '../../utilities/views';
+import { getCharts } from '../../utilities/vega-chart';
+
+/** NEEDED FOR NANOMINE */
+const LOCAL_DEV_SERVER = 'http://localhost:8000/nmr/chart';
+const SERVER = `${ROOT_URL}nmr/chart`;
+const URL = SERVER;
+
```

Comment:
This seems like a "god object".  Too many different things are defined in here and it could be nasty to maintain.  I would recommend finding a way to break it up.